### DISCUSSION TITLE
请问一下，这种类型的界面，如果用WPF来开发，里面的外框是用图片裁剪的还是用样式编写的？

### DISCUSSION THREAD

**Discussion by BeautyOfCoding** [BYJRK/DotNet-Discussions#99] [WPF 开发]
![MI9T{{6W_ 9)9~{LF~K`JQY](https://github.com/BYJRK/DotNet-Discussions/assets/46413316/6beedbe3-a8a9-4f5b-9a3b-9a5e43f5de93)
当然，我知道使用Border或者再添加一些样式肯定能实现这样的效果，但是感觉很麻烦。或者说这种边框是一种png或者svg图片吗，或者说没有有现成的框架能够直接拖拽出来这种效果？

**Comment by BYJRK** [comment]
这种界面如果在 Unity 里面制作，非常简单，因为可以设置一个类似于九宫格的东西，设定四个角保持缩放，然后四条边和中间是可以缩放的，这样就可以保证四个角不变的情况下组成任意大小的窗口。

如果要用 WPF 开发这样的界面，恐怕会麻烦不少，尤其是窗口大小可变的情况下。我可能会考虑的方式是分段绘制，比如 Header 用一个 Path，然后中间区域用一个高度可变的 Rectangle，并在上面放一个 ContentPresenter。它们各自控制自己的 Stroke 来绘制边框。最终它们组成了 Border 的 Template

**Reply by Dynesshely** [reply]
赞同

考虑为不同部分使用 svg 图片
从图上看, 这是一个简单的类窗口样式效果, 且宽度一定, 高度可变
我们可以拆分成 TitleBar, Border, Corner 三个部分
其中, TitleBar 包含标题, Corner 为四个角加上装饰
您可以使用一个 UserControl 包含这些组件

要想设计成宽高均可变的效果, 可以进行更加细致的拆分, 或者设计更好的 svg 图片或是使用 path 路径
